lib/customisation.nix

```diff
@@ -189,62 +190,76 @@ rec {
     );
 
   /**
-    Call the package function in the file `fn` with the required
-    arguments automatically.  The function is called with the
-    arguments `args`, but any missing arguments are obtained from
-    `autoArgs`.  This function is intended to be partially
-    parameterised, e.g.,
+    Call the function (or the function within the file) `funcOrPath` with `defaultArgs`, which can be overridden by `args`. This function is intended to be partially parameterised / curried (see examples).
+
+    # Inputs (positional)
+
+    - `defaultArgs` - `AttrSet`
+
+      Default
+
+    - `funcOrPath` - `(AttrSet -> a) | Path`
+
+      Either a function that is expected to build a package from the given attribute set or a file that returns such a function when imported.
+
+    - `args` - `AttrSet`
+
+      Arguments that will be passed to the function. These are merged with `defaultArgs`.
+      Attributes in `args` replace those in `defaultArgs`.
+
+      It is quite common to pass an empty attribute set if the `defaultArgs` suffice.
+
+    # Type
+
+    ```
+    callPackageWith :: AttrSet -> ((AttrSet -> a) | Path) -> AttrSet -> a
+    ```
+
+    # Examples
+
+    ## `pkgs.callPackage`
+
+    The most common use for this is the `pkgs.callPackage` function that uses `pkgs` as the `defaultArgs`.
+    See the documentation of that function for examples.
+
+    ## Contrived
+
+    :::{.example}
+    ## Create a custom `callPackage` function
+
+      This custom `callPackage` will call the function (or file with a function) that is passed to it with `libfoo` and `libbar` in the function arguments.
```

Comment:
Indentation seems off. Here and in other places. There is no indentation necessary. It would only wrap all the indented stuff here in another codeblock